### PR TITLE
sys/shell: added RAWMODE option to netif sc

### PR DESCRIPTION
**Pull request by haukepetersen**
edit:
~~now rebased on #2778 and #2845~~

### CODE REVIEW

**Comment by authmillenon** [comment]
I'd prefer it to be done as it is done in #2778 as it is compatible to the POSIX `ifconfig` command.

**Comment by authmillenon** [comment]
(see http://www.freebsd.org/cgi/man.cgi?ifconfig%288%29 for reference)

**Comment by haukepetersen** [comment]
I like this approach far better!

**Comment by authmillenon** [comment]
I like my approach better. A value-based syntax for flags is clunky, inelegant and I would argue with some optimization to the flag-based syntax in my approach even bigger in code-size.

File: sys/shell/commands/sc_netif.c [hunk 1]
```diff
@@ -332,6 +336,25 @@ static void _netif_set_state(kernel_pid_t dev, char *state_str)
...
+        state = NETCONF_DISABLE;
+    } else if (val[0] == '1') {
+        state = NETCONF_ENABLE;
+    } else {
+        puts("error: invalid value given");
+        return;
+    }
+    if (ng_netapi_set(dev, opt, 0, &state, sizeof(state)) < 0) {
+        puts("error: unable to set option\n");
+    } else {
```
Comment (Kijewski) [inline sys/shell/commands/sc_netif.c]: nl

**Comment by OlegHahm** [comment]
@authmillenon, can you send me a pointer to the POSIX spec for ifconfig?

**Comment by authmillenon** [comment]
Mh… though it seems to be shipped with every UNIX-oid operating system it does not seem to be part of POSIX. Regardless of that I would vote to implement it the way a typical Linux/BSD/MacOSX user would expect it to work.

**Comment by authmillenon** [comment]
(see `iproute2`/`ifconfig` split)

**Comment by OlegHahm** [comment]
For the record: I don't vote for the one or the other approach since I haven't really looked at it and I agree that code size is not really a big matter when it comes to the shell and I also agree that sticking to well-known syntax is a good idea.
However, I wonder how well-known and common `ifconfig` and its syntax still is. At least most people that I know are using and `iproute2` tools instead and I can hardly remember `ifconfig` syntax (even though I used it for almost a decade).

**Comment by OlegHahm** [comment]
(Although, I agree that the name `ifconfig` matches the task for this tool better than `ip`).

**Comment by authmillenon** [comment]
@OlegHahm `iproute2` is Linux only [1], FreeBSD and MacOSX are still sticking to `ifconfig` (which is why most Linux distros still ship it).

[1] http://www.linuxfoundation.org/collaborate/workgroups/networking/iproute2

**Comment by OlegHahm** [comment]
I'm definitely not voting for `iproute2` anyways. IMO the parser for that would be far too complex for RIOT.

**Comment by haukepetersen** [comment]
Actually my opinion about this is not too strong. So I will adapt this PR to the Syntax proposed by @authmillenon. In the end we need ASAP a working version of the network stack and it does not make sense to spend time on discussions like this one at the moment!

**Comment by authmillenon** [comment]
I think it's just the RAW flag you need to adapt.

**Comment by haukepetersen** [comment]
I would change all four flags to the [-]FLAG paradigm, right?

**Comment by haukepetersen** [comment]
oh, didn't see that you have the other ones already covered..

**Comment by authmillenon** [comment]
;-)

**Comment by haukepetersen** [comment]
updated. Rebased it on  #2778 and #2845.

**Comment by authmillenon** [comment]
Needs rebate.

**Comment by authmillenon** [comment]
(both dependencies are merged)

**Comment by haukepetersen** [comment]
dependencies are merged -> rebased.

File: sys/shell/commands/sc_netif.c [hunk 2]
```diff
@@ -430,6 +437,9 @@ static int _netif_flag(char *cmd, kernel_pid_t dev, char *flag)
     else if (strcmp(flag, "autoack") == 0) {
         return _netif_set_flag(dev, NETCONF_OPT_AUTOACK, set);
     }
+    else if (strcmp(flag, "raw"), == 0) {
```
Comment (authmillenon) [inline sys/shell/commands/sc_netif.c]: `s/),//`

**Comment by authmillenon** [comment]
Apart from https://github.com/RIOT-OS/RIOT/pull/2846#discussion_r28967250: ACK

**Comment by haukepetersen** [comment]
yapp, just saw it. Fixed.

**Comment by LudwigOrtmann** [comment]
@haukepetersen needs rebase

**Comment by haukepetersen** [comment]
rebased

**Pull request merged**